### ISSUE TITLE
Plugin does not enumerate any aircraft.

### ISSUE THREAD

**Issue by matt-cahill** [charlestytler/streamdeck-dcs-interface#4]
My installation directory is located at `D:\DCS World Openbeta` however when I change the path in the ID lookup window and click update, the plugin does not see any aircraft. From your comment on #1 I’m wondering if the plugin is not expecting an installation on another drive? I’ll try the plugin from the #3 PR and let you know how that goes. 

DCS v.2.5.6
Latest StreamDeck software
Version 1.0.1 plugin

**Comment by charlestytler** [comment]
Hi Matt,
It should support other drives, as I've tried that before. All it does is take your path, append `module_subdir="/mods/aircraft/"` (the forward slashes should still be fine in Windows) and query all files in that folder via:
`std::filesystem::directory_iterator(dcs_install_path + module_subdir)`

So it should be trying to get a `dir` listing of folders in `D:\DCS World Openbeta\mods\aircraft\`

I pushed a commit to the v1.0.2 branch that adds a log message if the path can't be found:
https://github.com/charlestytler/streamdeck-dcs-interface/commit/b1806768e2b8cb12895bfbee6eecd333881b0b2b

If you are on that branch you should be able to download the rebuilt `Release/com.ctytler.dcs.streamDeckPlugin`
and if the path isn't found you'll get a message logged within
`%appdata%\Elgato\StreamDeck\logs\com.ctytler.dcs0.log`
that should say something like:
`20:14:04.092 Get Installed Modules Failure: DCS Install path [D:\DCS World Openbeta/mods/aircraft/] not found.`

If you copy-paste what's inside the brackets of the error message into Windows Explorer does that path work?

Also, should there be a capital "B" in your path, `D:\DCS World OpenBeta`?

**Comment by matt-cahill** [comment]
Thanks @charlestytler. Yes `D:\DCS World OpenBeta` is the correct path but that was just a typo in this issue, I had the correct case sensitive path in the plugin though. 

Updating the plugin did not not help and also I did not get any logging which i though was strange and then this popped up:

![Annotation 2020-04-15 125952](https://user-images.githubusercontent.com/2563358/79288475-f2f15180-7f1a-11ea-90af-c6d5a4b4db05.png)

Turns out I was missing the Visual C runtime.

https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

Once installed the ID lookup works.

**Comment by charlestytler** [comment]
Wow good find, I'm surprised to see a dependency issue like this since I don't use much outside of the standard library for the C++ code. Although this is the first time I've tried to make a C++ program in a Windows environment, so I'm hoping this is rare.

Did you install a Creator's Update or have an older version of Windows or anything else you can think of that may be an issue?

Thanks for sharing the result though!

**Comment by matt-cahill** [comment]
Nothing that I can think of, it's just normal Windows 10 Pro fully up to date. I'm not sure why it would be missing that library if it is supposed to be installed by default. I do have another windows install (on a machine that couldn't possibly run DCS) that I can install stream deck and test on so i'll let you know if it just works there.

The wording here seems to infer they are not standard windows libraries though so it might be a bit hit and miss. If the user happens to have them from some other app already it will be ok.
https://docs.microsoft.com/en-us/cpp/windows/determining-which-dlls-to-redistribute?view=vs-2019

**Comment by matt-cahill** [comment]
I get the same missing library error on my other system and interestingly the warning dialog appeared immediately on installation this time.

This windows installation is a pretty much vanilla install. Windows 10 Pro, 1909, 18363.752

I have a feeling this will be fairly common but I'm not sure whether the plugin install process could take care of the dependency install as well? Perhaps it's just something to add to the docs?

**Comment by matt-cahill** [comment]
PS thanks for building this by the way, it looks like it's going to be awesome! Now I'm up and running I'm excited to start testing it out :)

**Comment by charlestytler** [comment]
Ok interesting, I did some research and using [this tool](https://github.com/lucasg/Dependencies) was able to get a list of DLL dependencies for my executable as well as Elgato's StreamDeck executable.
![image](https://user-images.githubusercontent.com/30204837/79330706-fe775500-7ee7-11ea-9a34-a40000d5bea1.png)
![image](https://user-images.githubusercontent.com/30204837/79330797-1ea71400-7ee8-11ea-806d-1f34ace448a0.png)

So it appears that two options are including a runtime installer with the program, or including dependent DLLs in the same directory as the executable as Elgato has done.

Thanks for identifying the issue. Perhaps other users so far have just conveniently had the runtime library installed from other software -- since it looks like Streamdeck isn't installing it.

**Comment by charlestytler** [comment]
I've added 760e6df51a80b017e497e563056ed35d34e49a53 to the v1.0.2 pull request #3 which includes the Visual C++ redistributables I believe are necessary for the program. It only adds ~200kB to the size of the Plugin installer.

While I'm not a big fan of including C++ runtime libraries, I think it's probably better than the alternative of requiring some users to install additional packages.

**Comment by charlestytler** [comment]
Following up on this, I was able to reproduce the issue on another Windows machine that had no Visual C++ Runtime newer than 2013 installed. Running the v1.0.2 branch resolved the issue.

I think I've tracked down the source of the dependency on newer versions to a new exception-handling feature (`__CxxFrameHandler4`) that is enabled by default in current Visual Studio versions and requires the additional `vcruntime140_1.dll` per https://devblogs.microsoft.com/cppblog/making-cpp-exception-handling-smaller-x64/

![image](https://user-images.githubusercontent.com/30204837/79399502-86de1000-7f51-11ea-8404-16e03f747e94.png)

**Comment by AMLARRONDO90** [comment]
Hello,

I followed your video on how to install and use the stream deck on DCS, I am getting DCS comms with my stream deck and I can even use some functions on my stream deck on DCS BUT, when I try to hot key buttons to a module (the F18) and have the DCS install directory set to where it should be no aircraft populate on the screen for me to choose. I have attached screen shots of what Im looking at. I am using the steam edition of DCS could this be the problem?
